EventEmitter.d.ts

```diff
@@ -19,30 +19,30 @@ declare namespace EventEmitter {
         [event: string]: Listener[];
     }
 
-    type Listeners = Listener[]|ListenerMap;
+    type Listeners = Listener[] | ListenerMap;
 
-    export class EventEmitter {
+    export class EventEmitter<E = EventKey> {
         static noConflict(): typeof EventEmitter;
 
-        getListeners(event: EventKey): Listeners;
+        getListeners(event: E): Listeners;
         flattenListeners(listeners: Listener[]): void;
-        getListenersAsObject(event: EventKey): ListenerMap;
-        addListener(event: EventKey, listener: Listener|Function): this;
-        on(event: EventKey, listener: Listener|Function): this;
-        addOnceListener(event: EventKey, listener: Function): this;
-        once(event: EventKey, listener: Function): this;
-        defineEvent(event: EventKey): this;
-        defineEvents(events: EventKey[]): this;
-        removeListener(event: EventKey, listener: Function): this;
-        off(event: EventKey, listener: Function): this;
+        getListenersAsObject(event: E): ListenerMap;
+        addListener(event: E, listener: Listener | Function): this;
+        on(event: E, listener: Listener | Function): this;
+        addOnceListener(event: E, listener: Function): this;
+        once(event: E, listener: Function): this;
+        defineEvent(event: E): this;
+        defineEvents(events: E[]): this;
+        removeListener(event: E, listener: Function): this;
+        off(event: E, listener: Function): this;
         addListeners(event: Events, listeners: Function[]): this;
         removeListeners(event: Events, listeners: Function[]): this;
         manipulateListeners(remove: boolean, event: Events, listeners: Function[]): this;
-        removeEvent(event?: EventKey): this;
-        removeAllListeners(event?: EventKey): this;
-        emitEvent(event: EventKey, args?: any[]): this;
-        trigger(event: EventKey, args?: any[]): this;
-        emit(event: EventKey): this;
```

Comment:
Would this actually work though, like if you said "my event key type is now MyType", you could then call `.addEventListener(somethingOfMyType, fn)`, which would type check, but throw an exception. Right? You can pass any type you want as a key, so I'm not sure how this added functionality helps.

Had a look, yeah, whatever you pass is expected to either be a regular expression or a string. Anything else will be coerced to a string. I don't see the benefit for letting users change the key type. The only thing that should type check should be a string or a regular expression, as is the case right now.

https://github.com/Olical/EventEmitter/blob/6f211f4224100ad505d7e8ef65ab13093ccbb9db/EventEmitter.js#L80